keadm/cmd/keadm/app/cmd/edge/joinbeta.go

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright 2022 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
+	"sigs.k8s.io/yaml"
+
+	"github.com/kubeedge/kubeedge/common/constants"
+	"github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/common"
+	"github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/util"
+	"github.com/kubeedge/kubeedge/pkg/apis/componentconfig/edgecore/v1alpha1"
+	"github.com/kubeedge/kubeedge/pkg/apis/componentconfig/edgecore/v1alpha1/validation"
+	"github.com/kubeedge/kubeedge/pkg/image"
+	pkgutil "github.com/kubeedge/kubeedge/pkg/util"
+	"github.com/kubeedge/kubeedge/pkg/version"
+)
+
+var (
+	edgeJoinBetaDescription = `
+"keadm beta join" command bootstraps KubeEdge's worker node (at the edge) component.
+It will also connect with cloud component to receive
+further instructions and forward telemetry data from
+devices to cloud
+`
+	edgeJoinBetaExample = `
+keadm beta join --cloudcore-ipport=<ip:port address> --edgenode-name=<unique string as edge identifier>
+
+  - For this command --cloudcore-ipport flag is a required option
+  - This command will download and install the default version of pre-requisites and KubeEdge
+
+keadm beta join --cloudcore-ipport=10.20.30.40:10000 --edgenode-name=testing123 --kubeedge-version=v` + common.DefaultKubeEdgeVersion + `
+`
+)
+
+func NewJoinBetaCommand() *cobra.Command {
+	joinOptions := newOption()
+	step := common.NewStep()
+	cmd := &cobra.Command{
+		Use:          "join",
+		Short:        "Bootstraps edge component. Checks and install (if required) the pre-requisites. Execute it on any edge node machine you wish to join",
+		Long:         edgeJoinBetaDescription,
+		Example:      edgeJoinBetaExample,
+		SilenceUsage: true,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			step.Printf("Check KubeEdge edgecore process status")
+			running, err := util.IsKubeEdgeProcessRunning(util.KubeEdgeBinaryName)
+			if err != nil {
+				klog.Exitf("Check KubeEdge edgecore process status failed: %v", err)
+			}
+			if running {
+				klog.Exitln("EdgeCore is already running on this node, please run reset to clean up first")
+			}
+
+			step.Printf("Check if the management directory is clean")
+			if _, err := os.Stat(joinOptions.TarballPath); err != nil {
```

Comment:
The `TarballPath` is used to store the release tar in previous version, I think it's no needed in new version, we only have the docker image and binary/configfile, Can we delete it or replace other option? :)